scripts/amex_browseros_bridge.py

```
@@ -0,0 +1,543 @@
+) -> None:
+    deadline = time.monotonic() + ACTION_TIMEOUT_SECONDS
+    while time.monotonic() < deadline:
+        current = await snapshot(session, page)
+        require_amex_url(current)
+        if is_dashboard_ready(current):
+            return
+        if 'textbox "User ID"' in current:
+            await authenticate(session, page, user, password, current)
+            return
+        await asyncio.sleep(0.5)
+    raise RuntimeError("Timed out waiting for Amex dashboard or login page")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)